events/janus_gelfevh.c

```diff
@@ -0,0 +1,590 @@
+const char *janus_gelfevh_get_package(void);
+void janus_gelfevh_incoming_event(json_t *event);
+json_t *janus_gelfevh_handle_request(json_t *request);
+
+/* Event handler setup */
+static janus_eventhandler janus_gelfevh =
+	JANUS_EVENTHANDLER_INIT (
+		.init = janus_gelfevh_init,
+		.destroy = janus_gelfevh_destroy,
+
+		.get_api_compatibility = janus_gelfevh_get_api_compatibility,
+		.get_version = janus_gelfevh_get_version,
+		.get_version_string = janus_gelfevh_get_version_string,
+		.get_description = janus_gelfevh_get_description,
+		.get_name = janus_gelfevh_get_name,
+		.get_author = janus_gelfevh_get_author,
+		.get_package = janus_gelfevh_get_package,
+
+		.incoming_event = janus_gelfevh_incoming_event,
+		.handle_request = janus_gelfevh_handle_request,
+
+		.events_mask = JANUS_EVENT_TYPE_NONE
+	);
+
+/* Plugin creator */
+janus_eventhandler *create(void) {
+	JANUS_LOG(LOG_VERB, "%s created!\n", JANUS_GELFEVH_NAME);
+	return &janus_gelfevh;
+}
+
+/* Compression, if any */
+static gboolean compress = FALSE;
+static int compression = 6; /* Z_DEFAULT_COMPRESSION */
+
+/* Useful stuff */
+static volatile gint initialized = 0, stopping = 0;
+static GThread *handler_thread;
+static void *janus_gelfevh_handler(void *data);
+static janus_mutex evh_mutex;
+
+/* JSON serialization options */
+static size_t json_format = JSON_INDENT(3) | JSON_PRESERVE_ORDER;
+
+/* Queue of events to handle */
+static GAsyncQueue *events = NULL;
+static json_t exit_event;
+static void janus_gelfevh_event_free(json_t *event) {
+	if(!event || event == &exit_event)
+		return;
+	json_decref(event);
+}
+
+/* GELF backend to send the events to */
+static char *backend = NULL;
+static char *port = NULL;
+
+typedef enum janus_gelfevh_socket_type {
+	JANUS_GELFEVH_SOCKET_TYPE_TCP = 1,
+	JANUS_GELFEVH_SOCKET_TYPE_UDP = 2
+} janus_gelfevh_socket_type;
+
+static int max_gelf_msg_len = 500;
+static int sockfd;
+/* Set TCP as Default transport */
+static janus_gelfevh_socket_type transport = JANUS_GELFEVH_SOCKET_TYPE_UDP;
+
+/* Parameter validation (for tweaking via Admin API) */
+static struct janus_json_parameter request_parameters[] = {
+	{"request", JSON_STRING, JANUS_JSON_PARAM_REQUIRED}};
+static struct janus_json_parameter tweak_parameters[] = {
+	{"events", JSON_STRING, 0},
+	{"backend", JSON_STRING, 0},
+	{"port", JSON_STRING, 0},
+	{"max_gelf_msg_len", JSON_INTEGER, JANUS_JSON_PARAM_POSITIVE},
+	{"janus_gelfevh_socket_type", JSON_STRING, 0}
+};
+/* Error codes (for the tweaking via Admin API */
+#define JANUS_GELFEVH_ERROR_INVALID_REQUEST		411
+#define JANUS_GELFEVH_ERROR_MISSING_ELEMENT		412
+#define JANUS_GELFEVH_ERROR_INVALID_ELEMENT		413
+#define JANUS_GELFEVH_ERROR_UNKNOWN_ERROR		499
+
+/* Plugin implementation */
+static char *randstring(size_t length) {
+	static char charset[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+	char *randomString = NULL;
+	if (length) {
+		randomString = malloc(sizeof(char) * (length + 1));
```

Comment:
This should be `g_malloc`, because you're using `g_free` to free it later on. Using `g_free` on something not allocated by glib will likely cause issues.